agent/stats/engine.go

```diff
@@ -123,20 +164,15 @@ func (engine *DockerStatsEngine) MustInit(taskEngine ecsengine.TaskEngine, clust
 		return err
 	}
 
-	return engine.Init()
-}
-
-// Init subscribes to the container change event stream.
-func (engine *DockerStatsEngine) Init() error {
 	// Subscribe to the container change event stream
-	err := engine.containerChangeEventStream.Subscribe(containerChangeHandler, engine.handleDockerEvents)
+	err = engine.containerChangeEventStream.Subscribe(containerChangeHandler, engine.handleDockerEvents)
 	if err != nil {
 		return fmt.Errorf("Failed to subscribe to container change event stream, err %v", err)
 	}
 
-	go engine.listContainersAndStartEventHandler()
-	go engine.waitToStop()
+	engine.synchronizeState()
```

Comment:
this returns an error. please make sure that you log a warning if an error is returned.